breakpad-symbols/src/sym_file/parser.rs

```diff
@@ -390,23 +415,16 @@ impl SymbolParser {
 
             // Parse a top-level item, and first handle the Result
             let line = match line(input) {
-                Done(new_input, line) => {
+                Ok((new_input, line)) => {
                     // Success! Advance the input.
                     input = new_input;
                     line
                 }
-                Error(_) => {
+                Err(_) => {
                     // The file has a completely corrupt line,
                     // conservatively reject the entire parse.
                     return Err(SymbolError::ParseError("failed to parse file", self.lines));
                 }
-                Incomplete(_) => {
-                    // One of our sub-parsers wants more input, which normally
-                    // would be fine for a streaming parser, bust the newline
-                    // preprocessing we do means this should never happen.
-                    // So Incomplete input is just another kind of parsing Error.
-                    return Err(SymbolError::ParseError("line was incomplete!", self.lines));
-                }
```

Comment:
good call. The Incomplete stuff was an artifact of me trying to use the streaming machinery and then realizing it didn't actually work for this problem.

Reply:
Oh I wasn't thinking any smart thoughts about this, I was just forced into this change by the type system, because nom's "complete" parsers can no longer return an Incomplete status.